src/keycloak/chart/templates/_helpers.tpl

```diff
@@ -74,3 +74,23 @@ Create the service DNS name.
 {{- define "keycloak.serviceDnsName" -}}
 {{ include "keycloak.fullname" . }}-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
 {{- end }}
+
+{{/*
+Check external PostgreSQL connection information. Fails when required values are missing or if PostgreSQL is configured when devMode is enabled.
+*/}}
+{{- define "keycloak.postgresql.config" -}}
+{{- if not .Values.devMode -}}
+{{- if .Values.postgresql -}}
+{{ $requiredKeys := list "username" "password" "database" "host" "port" }}
+{{- range $k := $requiredKeys -}}
+{{ if empty (get $.Values.postgresql $k) }}{{- fail (printf "Missing value for \"postgresql.%s\"." $k ) -}}{{- end }}
+{{- end }}
+{{- else -}}{{fail "You must define \"username\", \"password\", \"database\", \"host\", and \"port\" for \"postgresql\"."}}
+{{- end -}}
+{{- default "true" "" }}
+{{- else if not (empty (compact (values .Values.postgresql))) -}}
+{{ fail "Cannot use an external PostgreSQL Database when devMode is enabled." -}}
```

Comment:
Adding a check for `.Values.postgresql.enabled` here would allow users to easily toggle between `devMode` and using `postgresql` if desired. Currently, users have to omit all values under `.Values.postgresql` to successfully enable `devMode`.

Thoughts on adding `postgresql.enabled`?

Reply:
Hm, I think that's viable. Can we easily do this in a non-breaking change way, that wouldn't be confusing 😆?

In the past to "enable postgres" I just needed to set `devMode: false`, technically (although I would also obviously need the actual postgres values specified for it to work. If we default `postgresql.enabled` to false, then my upgrade would result in these values:

```yaml
devMode: false
postgresql:
  enabled: false
```

If this still enabled postgres then it would be non-breaking, but that might be confusing? I think I'd be happy with leaving off the explicit `enabled` key and using our multi-key check/logic, as long as we have those clear `fail` messages. Open to other opinions though, I think to some extent our `devMode` key is already the inverse of `postgresql.enabled` so having both feels duplicative to me (if we wanted to add an `enabled` key it might make sense to deprecate/remove the `devMode` one).

Reply:
Chatted offline with @mjnagel, we decide to leave this out for now.